specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2024-09-01/DataConnectors.json

```diff
@@ -0,0 +1,1696 @@
+          "type": "object"
+        }
+      },
+      "type": "object"
+    },
+    "AwsCloudTrailDataConnector": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/DataConnector"
+        }
+      ],
+      "description": "Represents Amazon Web Services CloudTrail data connector.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/AwsCloudTrailDataConnectorProperties",
+          "description": "Amazon Web Services CloudTrail data connector properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object",
+      "x-ms-discriminator-value": "AmazonWebServicesCloudTrail"
+    },
+    "AwsCloudTrailDataConnectorDataTypes": {
+      "description": "The available data types for Amazon Web Services CloudTrail data connector.",
+      "properties": {
+        "logs": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/DataConnectorDataTypeCommon"
+            }
+          ],
+          "description": "Logs data type.",
+          "type": "object"
+        }
+      },
+      "type": "object"
+    },
+    "AwsCloudTrailDataConnectorProperties": {
+      "description": "Amazon Web Services CloudTrail data connector properties.",
+      "properties": {
+        "awsRoleArn": {
+          "description": "The Aws Role Arn (with CloudTrailReadOnly policy) that is used to access the Aws account.",
+          "type": "string"
+        },
+        "dataTypes": {
+          "$ref": "#/definitions/AwsCloudTrailDataConnectorDataTypes",
+          "description": "The available data types for the connector."
+        }
+      },
+      "type": "object"
+    },
+    "RestApiPollerDataConnector": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/DataConnector"
+        }
+      ],
+      "description": "Represents Rest Api Poller data connector.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/RestApiPollerDataConnectorProperties",
+          "description": "Rest Api Poller data connector properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object",
+      "x-ms-discriminator-value": "RestApiPoller"
+    },
+    "RestApiPollerDataConnectorProperties": {
+      "description": "Rest Api Poller data connector properties.",
+      "properties": {
+        "connectorDefinitionName": {
+          "description": "The connector definition name (the dataConnectorDefinition resource id).",
+          "type": "string"
+        },
+        "auth": {
+          "description": "The a authentication model.",
+          "$ref": "#/definitions/CcpAuthConfig"
+        },
+        "request": {
+          "description": "The request configuration.",
+          "$ref": "#/definitions/RestApiPollerRequestConfig"
+        },
+        "dcrConfig": {
+          "description": "The DCR related properties.",
+          "$ref": "#/definitions/DCRConfiguration"
+        },
+        "isActive": {
```

Comment:
>isActive [](http://example.com/codeflow?start=9&length=8)

You should convert this to an enum 

Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].